### DISCUSSION TITLE
Complex API rules

### DISCUSSION THREAD

**Discussion by charbs-io** [pocketbase/pocketbase#2213] [Q&A] [answered]
Hi, 

I have the following collections:

Users 
Workspaces 
Workspace_users 
Posts

A user can belong to many workspaces. The workspace collection only holds data on the workspace plan and subscription. 

The workspace_users collection is then responsible for assigning users to workspaces, as well as assigning a role for the given user. This collection has a 1:1 relation with both users collection and the workspaces collection. 

The posts collection holds data on a post, as well as the workspace it belongs to and the user that created it. 

I'm struggling to wrap my head around the api rules ui in admin to come up with a rule to fetch posts only if a workspace_users record matches the workspace of the post.

I guess what I am trying to do is find the row in workspace_users, match its users ID with the request auth ID AND ensure the post workspace ID matches the workspace ID from that same row. 

I'm not near a computer anymore but will provide examples of what I have tried to do to no avail. 

Any help would be appreciated!

**Comment by ganigeorgiev** [comment, marked as answer]
Because there is no direct relation to the `workspace_users` from the `posts` or the `users` collections in this case you would have to use the `@collection.*` filter. It acts as a SQL join and can be used to join any other collection (including the current/main itself) based on the user provided conditions. Something like:

```js
@collection.workspace_users.workspace ?= workspace &&
@collection.workspace_users.user ?= @request.auth.id
```
_(aka. must have at least one `workspace_users` entry where the "workspace" field matches with the post workspace and the "user" field matches with the id of the current authenticated user)_

This eventually will be simplified in the near future with the planned "indirect relations" filter and sort (_similar to the [indirect expand](https://pocketbase.io/docs/expanding-relations/#indirect-expand)_).

**Reply by charbs-io** [reply]
Wow, i had no idea `@collection` acts as a SQL join and I also had no idea about the indirect expand option. Thanks!
